src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/multitable/MultitableFlowsSpec.groovy

```diff
@@ -136,237 +135,188 @@ mode with existing flows and hold flows of different table-mode types"() {
+        given: "An active switch"
+        def sw = topology.activeSwitches.find { it.features.contains(SwitchFeature.MULTI_TABLE) }
+        SwitchPropertiesDto initSwProps = switchHelper.getCachedSwProps(sw.dpId)
 
-        then: "Flow rules on the switch are not recreated in single table mode because flow wasn't rerouted"
-        verifyAll(northbound.getSwitchRules(sw.dpId).flowEntries) { rules ->
-            rules.find { it.cookie == flowInfoFromDb3.forwardPath.cookie.value }.tableId == INGRESS_RULE_MULTI_TABLE_ID
-            rules.find { it.cookie == flowInfoFromDb3.reversePath.cookie.value }.tableId == INGRESS_RULE_MULTI_TABLE_ID
-            rules.findAll { it.cookie in sharedRules*.cookie }*.tableId.unique() == [SHARED_RULE_TABLE_ID]
+        and: "Multi table mode is disabled on it"
+        if (initSwProps.multiTable) {
+            northbound.updateSwitchProperties(sw.dpId, changeSwitchPropsMultiTableValue(initSwProps, false))
         }
+        checkDefaultRulesOnSwitch(sw)
 
-        when: "Reroute(intentional) the flow via APIv2"
-        with(northboundV2.rerouteFlow(flow.flowId)) { !it.rerouted }
-        wait(rerouteDelay + WAIT_OFFSET) {
-            assert northboundV2.getFlowStatus(flow.flowId).status == FlowState.UP
-        }
+        when: "Create a single-switch flow"
+        def flow = flowHelperV2.singleSwitchFlow(sw)
+        flowHelperV2.addFlow(flow)
 
-        then: "Flow rules on the switch are not recreated in single table mode because the flow wasn't rerouted"
-        verifyAll(northbound.getSwitchRules(sw.dpId).flowEntries) { rules ->
-            rules.find { it.cookie == flowInfoFromDb3.forwardPath.cookie.value }.tableId == INGRESS_RULE_MULTI_TABLE_ID
-            rules.find { it.cookie == flowInfoFromDb3.reversePath.cookie.value }.tableId == INGRESS_RULE_MULTI_TABLE_ID
-        }
+        then: "Flow rules are created in single table mode"
+        assert "flow rules are in tables"(sw, flow) == [SINGLE_TABLE_ID]
 
         and: "Flow is valid"
         northbound.validateFlow(flow.flowId).each { direction -> assert direction.asExpected }
 
-        when: "Delete the flow"
-        flowHelperV2.deleteFlow(flow.flowId)
-        def flowIsDeleted = true
+        when: "Update switch properties(multi_table: true) on the switch"
+        northbound.updateSwitchProperties(sw.dpId, changeSwitchPropsMultiTableValue(initSwProps, true))
 
-        then: "Flow rules are deleted"
-        northbound.getSwitchRules(sw.dpId).flowEntries.findAll {
-            def cookie = new Cookie(it.cookie)
-            cookie.type == CookieType.MULTI_TABLE_INGRESS_RULES || !cookie.serviceFlag
-        }.empty
+        and: "reroute the flow"
+        with(northboundV2.rerouteFlow(flow.flowId)) { !it.rerouted }
 
+        then: "Flow rules stay in single table mode"
+        wait(RULES_INSTALLATION_TIME/5) {
+            "flow rules are in tables"(sw, flow) == [SINGLE_TABLE_ID]
```

Comment:
why `/5` ?
What if, because of bug, rules will be actually reinstalled after `RULES_INSTALLATION_TIME` but we will miss it because we waited not enough?

Reply:
Originally there was `/3`, but you are right, we should wait for the whole time.